lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -1535,6 +1550,8 @@ ekn_media_bin_handle_msg_state_changed (EknMediaBin *self, GstMessage *msg)
              gst_element_state_get_name (old_state),
              gst_element_state_get_name (new_state));
 
+  ekn_media_bin_dump_dot (self, old_state, new_state);
```

Comment:
Don't you want to put some ifdef or env var guard around this so that you're not computing a filename on every state change?